primary/src/grpc_server.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
On a more pedestrian topic:
- the grpc server will only be run in the case Narwhal is coupled with an external consensus algorithm,
- it only serves the interface with that consensus, 
- the optionality of running it should be present in setup (the optionality in running tusk is a good place to look in order to see an example)

Reply:
Would it be fair to assume that if the [`consensus`](https://github.com/MystenLabs/narwhal/blob/main/node/src/lib.rs#L96) is set to false then we are planning to run narwhal with an external consensus algorithm? So I can reuse the `consensus` bool to see if I should start the gRPC server or not?

Or should I make it a separate subcommand as part of launching a primary?

Reply:
> Would it be fair to assume that if the [consensus](https://github.com/MystenLabs/narwhal/blob/main/node/src/lib.rs#L96) is set to false then we are planning to run narwhal with an external consensus algorithm?

Yes, totally.